lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb

```diff
@@ -606,47 +606,76 @@ def convert_to_service_component(service_branch)
       # @param approximate_date_source Hash EVSS format data object
       # @param short boolean (optional) return a shortened date YYYY-MM
       def convert_approximate_date(approximate_date_source, short: false)
-        approximate_date = approximate_date_source['year'].to_s
-        approximate_date += "-#{approximate_date_source['month']}" if approximate_date_source['month']
+        year  = approximate_date_source['year'].to_s.strip
+        month = approximate_date_source['month'].to_s.strip
+        day   = approximate_date_source['day'].to_s.strip
 
-        # returns YYYY-MM if only "short" date is requested
-        return approximate_date if short
+        # Treat blanks and "XX" placeholders as absent
+        month = nil if month.blank? || month == 'XX'
+        day   = nil if day.blank?   || day == 'XX'
+
+        approximate_date = year
+        approximate_date += "-#{month}" if month
 
-        approximate_date += "-#{approximate_date_source['day']}" if approximate_date_source['day']
+        return approximate_date if short
 
+        approximate_date += "-#{day}" if day
         approximate_date
       end
 
       def transform_disabilities(disabilities_source, toxic_exposure_conditions)
-        disabilities_source.map do |disability_source|
+        tox_present = toxic_exposure_conditions.present?
```

Comment:
```suggestion
        tox_present = toxic_exposure_conditions.present? && toxic_exposure_conditions.any?
```
to double-check: do we not need this 2nd condition? i see it in the default branch ([ref](https://github.com/department-of-veterans-affairs/vets-api/blob/c2969930c4a2c4f195c39682352cf1df020104db/lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb#L625-L627)) , but not in this PR.

Reply:
@lisac present and any are doing the same thing in different ways. It would be great to get a second reviewer perspective.